core/src/test/java/org/elasticsearch/action/update/UpdateRequestTests.java

```diff
@@ -322,6 +353,38 @@ public void testNowInScript() throws IOException {
         }
     }
 
+    public void testTimeout() {
+        final GetResult getResult =
+                new GetResult("test", "type", "1", 0, true, new BytesArray("{\"f\":\"v\"}"), null);
+
+        final boolean delete = randomBoolean();
+        final Script script =
+                delete ? mockInlineScript("ctx.op = delete") : mockInlineScript("return");
+        final UpdateRequest updateRequest =
+                new UpdateRequest("test", "type", "1")
+                        .script(script)
+                        .timeout(randomTimeValue());
+        final UpdateHelper.Result result = updateHelper.prepare(
+                new ShardId("test", "", 0),
+                updateRequest,
+                getResult,
+                ESTestCase::randomNonNegativeLong);
+        final Streamable action = result.action();
+        if (delete) {
+            assertThat(action, instanceOf(DeleteRequest.class));
```

Comment:
I tend to prefer letting the test throw the ClassCastException rather than asserting this first.